Add docs for Speculation Rules features: target_hint and clear-site-data (#39652)
mdn/content: files/en-us/web/api/speculation_rules_api/index.md

The best mitigation for these cases, and indeed any time when content can get out of sync with the server, is for pages to refresh themselves as needed. For example, a server might use the [Broadcast Channel API](/en-US/docs/Web/API/Broadcast_Channel_API), or another mechanism such as {{domxref("Window/fetch", "fetch()")}} or a {{domxref("WebSocket")}}. Pages can then update themselves appropriately, including speculatively loaded pages that have not yet been activated.

> Stale speculations can be cleared using the {{ httpheader("Clear-Site-Data") }} response header with the {{ httpheader("Clear-Site-Data#prefetchCache", `prefetchCache`) }} or {{ httpheader("Clear-Site-Data#prerenderCache", `prerenderCache`) }} values (or both) as appropriate where keeping them fresh is not possible. This header can be returned on any same-site HTTP request (such as an `/api/add-to-cart` API call).

Reviewer: You mean keep the prefetch part as discussed in https://github.com/mdn/content/pull/39652/files#r2108088582 ?
That is worth doing. 

The main point though I was trying to make here was to show the concrete case where all those prerendered pages might be "invisibly displaying the number of items in the cart". it just feels more "real".
I'll keep an eye out for your changes.

> Stale speculations can be cleared using the {{ httpheader("Clear-Site-Data") }} response header with the {{ httpheader("Clear-Site-Data#prefetchCache", `prefetchCache`) }} or {{ httpheader("Clear-Site-Data#prerenderCache", `prerenderCache`) }} values (or both) as appropriate where keeping them fresh is not possible. This header can be returned on any same-site HTTP request (such as an `/api/add-to-cart` API call).

Author: I meant the other changes you suggested here.

I've just realise the add to cart example follows on from the previous example. So don't think the middle paragraph is needed as it's repetitive so went with this:

```suggestion
Where refreshes are not possible, speculations can be cleared using the {{ httpheader("Clear-Site-Data") }} response header with the {{ httpheader("Clear-Site-Data#prefetchCache", `prefetchCache`) }} or {{ httpheader("Clear-Site-Data#prerenderCache", `prerenderCache`) }} values (or both) as appropriate.

The header can be returned on any same-site HTTP request (such as an `/api/add-to-cart` API call).
```
